Clients/Xamarin.Interactive.Client/Telemetry/Client.cs

```diff
@@ -6,92 +6,69 @@
-                            .SystemProcessInfo
-                            .GetAllProcesses ()
-                            .FirstOrDefault (proc =>
-                                 proc.ExecPath.EndsWith (
-                                     "dotnet",
-                                     StringComparison.Ordinal) &&
-                                 proc.Arguments.Any (a => a.EndsWith (
-                                     "/Xamarin.Interactive.Telemetry.Server.dll",
-                                     StringComparison.Ordinal)));
-
-                        if (serverProc != null)
-                            // FIXME: server process to provide endpoint somehow
-                            telemetryEndpoint = "http://localhost:5000/api/";
-                    } catch {
-                    }
-                }
-
-                if (!Prefs.Telemetry.Enabled.GetValue () ||
-                    !Uri.TryCreate (telemetryEndpoint, UriKind.Absolute, out var telemetryApiUri)) {
+                if (!Prefs.Telemetry.Enabled.GetValue ()) {
                     Log.Info (TAG, "Telemetry is disabled");
                     return;
                 }
 
                 enabled = true;
 
-                httpClient = new HttpClient {
-                    BaseAddress = telemetryApiUri
+                appInsightsClient = new TelemetryClient ();
+                appInsightsClient.InstrumentationKey = "@TELEMETRY_INSTRUMENTATION_KEY@";
+                appInsightsClient.Context.Session.Id = sessionid.ToString ();
+                appInsightsClient.Context.Device.OperatingSystem = host.OSName.ToString ();
+
+                var commonProperties = new Dictionary<string, string> {
+                    { "Product Version", BuildInfo.VersionString },
+                    { "Build Hash", BuildInfo.Hash },
+                    { "OS Name", host.OSName.ToString () },
+                    { "OS Version", host.OSVersion.ToString () },
+                    { "Word Size", IntPtr.Size.ToString () },
+                    { "CPU Word Size", (Environment.Is64BitOperatingSystem ? 8 : 4).ToString () },
+                    { "Release Candidate Level", ((byte) BuildInfo.Version.CandidateLevel).ToString () },
+                    { "Release Candidate Level Name", BuildInfo.Version.CandidateLevel.ToString ().ToLowerInvariant () },
+                    { "Machine ID", Sha256Hasher.Hash (MacAddressGetter.GetMacAddress ())}
                 };
 
-                httpClient.DefaultRequestHeaders.UserAgent.Add (
-                    new ProductInfoHeaderValue (
-                        "XamarinInteractiveTelemetry",
-                        BuildInfo.VersionString));
-            } catch (Exception e) {
-                Log.Error (TAG, "Unable to create HttpClient for telemetry", e);
-            }
-        }
+                foreach (var pair in commonProperties)
+                    appInsightsClient.Context.Properties.Add (pair);
```

Comment:
`TelemetryContext.Properties` is obsolete (at least in master), you want `TelemetryContext.GlobalProperties` it seems? Eliminate the extra dictionary work by just doing (annoying, I know):

```csharp
var globalProperties = appInsightsClient.Context.GlobalProperties;
globalProperties.Add ("Product Version", BuildInfo.VersionString);
globalProperties.Add (...);
...
```

Remember to add update channel here as well.

Reply:
`GlobalProperties` is in a newer beta that just came out, will bump to it.

Reply:
They're going to change the implementation again (https://github.com/Microsoft/ApplicationInsights-dotnet/issues/820#issuecomment-393909974) but I'll still make this change for the sake of clarity.

The linked issue also explains that right now, there's no difference in what goes to the server, but they plan to change that in the future (which will be nice when their UI can differentiate global properties).

Reply:
`GlobalProperties` release hasn't hit NuGet yet.